scripts/katana_traversal/traversal.py

```diff
@@ -0,0 +1,97 @@
+from abc import ABC, abstractmethod
```

Comment:
I'm pretty sure everything you have here is already available in a functional form in the standard library. Maybe in `shutil` or `glob` I'm not sure. Regardless, I'm pretty sure we don't need visitors here. They are a real pain compared to using generators.

Reply:
The visitor is very easy to use: just inherit from the base class and override the methods. On the other hand, I have only a limited understanding of Python generators, but as I understand we are not generating anything except the `bool`s returned, and we need to pass a parameter in every visiting step, rather than initializing the generator once and then getting values from it.